source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -667,6 +678,12 @@ The event API here is assumed to be like the standard `Python Event
                 # Wait before running the next check.
                 wait()
                 continue
+
+            # for drivers that implement CMAP, mark the connection pool as ready
+            # once the server transitions to a known state.
+            if previousDescription.type == Unknown and description.type != Unknown:
+                mark connection pool for server as "ready"
```

Comment:
I think it would be safer to mark the pool ready every time the server is in a known state:
```
            if description.type != Unknown:
                mark connection pool for server as "ready"
```

This would be a safeguard in case a driver has a bug where they pause the pool without marking the server unknown. In the current design such a bug would result in the pool being paused forever. With this change, the pool would only be paused for at most heartbeatFrequencyMS.

We should update the prose above to remove "`that was previously
Unknown`".

Reply:
done